src/client/proxy/handlers/common.ts

```diff
@@ -409,11 +426,11 @@ const getHttpRequestHandler = () => async (req: FastifyRequest, res: FastifyRepl
                     });
                 }
             }
-            res.status(404).send('Not Found');
+            return res.status(404).send('Not Found');
         }
     } catch (e) {
         log.error({stack: e.stack, errorMessage: e.message}, 'Proxy error');
-        res.status(500).send('Internal engine error');
+        return res.status(500).send('Internal engine error: ' + JSON.stringify(e.message));
```

Comment:
Normally, server should not expose internal error to the client. However, our "server" is not a regular one, it's always local. But anyway, do you think it's needed? In case of development, we can always see the point node logs. In case of a bug, the point node logs go to logstash. But do regular users need these messages?

Reply:
Three groups of people
- users don't care about errors
- we can look it up in logs
- but developers, when in dapp development mode, would be able to immediately see if e.g. zhtml template is faulty. It was very annoying to switch to console output each time.

Maybe later we can consult the mode and see if we're in dapp dev mode and only then show the error

But also consider the situations where users encounter an error in a dApp and complain to the dApp development team. In that case, it would be much better for them to send the full error instead of just "Internal error" (and users sending logs to external dapp dev teams can be privacy-breaking)